### PR TITLE
AWS: Add socket connection timeout for UrlConnectionHttpClient

### PR DESCRIPTION
**Pull request by JonasJ-ap**
User can use `http-client.urlconnection.socket-timeout-ms` and `http-client.urlconnection.connection-timeout-ms` tags to configure the connection and socket timeout for the `UrlConnectionHttpClient`.

Add unit test to check the timeout setting.

### CODE REVIEW

**Review by jackye1995 — APPROVED** [review]
looks good to me!

**Review by Fokko — APPROVED** [review]

**Comment by JonasJ-ap** [comment]
Thank you for your reviews! @rajarshisarkar @singhpk234 @amogh-jahagirdar May I ask if there is anything else you want to comment on this change? Thank you in advance.

File: aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java [hunk 1]
```diff
@@ -517,6 +539,8 @@ public class AwsProperties implements Serializable {
   public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
 
   private String httpClientType;
+  private Long httpClientUrlConnectionConnectionTimeoutMs;
+  private Long httpClientUrlConnectionSocketTimeoutMS;
```
Comment (singhpk234) [inline aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:543]: [nit] let make this `Ms`
```suggestion
  private Long httpClientUrlConnectionSocketTimeoutMs;
```

**Review by singhpk234 — APPROVED** [review]
LGTM as well !, Thanks @JonasJ-ap

**Comment by jackye1995** [comment]
Looks like we get a few approvals, thanks @JonasJ-ap for the contribution and thanks @Fokko and @singhpk234 for review!

**Pull request merged**